subsys/bluetooth/host/hci_core.c

```diff
@@ -956,7 +965,7 @@ static void conn_auto_initiate(struct bt_conn *conn)
 	    BT_FEAT_LE_PHY_2M(bt_dev.le.features)) {
 		err = bt_le_set_phy(conn, 0U, BT_HCI_LE_PHY_PREFER_2M,
 				    BT_HCI_LE_PHY_PREFER_2M,
-				    BT_HCI_LE_PHY_CODED_ANY);
+				    BT_HCI_LE_PHY_CODED_ANY, false);
```

Comment:
I think they should all use send_sync.
If you ignore the HCI error code here, then you wouldn't go to the data length update procedure below. Since that would only continue when the phy update complete event arrives.

Reply:
@joerchan , I'm not sure if I fully understand the problem. I changed the command to always be synchronous.

Reply:
`hci_le_read_remote_features` is not using the send_sync variant.
If it returns an HCI error code it would lead to the rest of the procedures not being done.

Reply:
Fixed by issuing all commands synchronously. That brings in other advantages as well.